### DISCUSSION TITLE
add a RSU to SIMU5G to communicate with eNodeB

### DISCUSSION THREAD

**Discussion by Y-AMDIAZ** [Unipisa/Simu5G#197] [Q&A]
I want to incorporate RSU (Roadside Unit) into my simulation simu5G to receive messages from cars and send them to eNodeB1. However, I'm encountering challenges when adding it, and I'm seeking guidance on where and how to modify the setup to facilitate communication. Can you please help?
<img width="784" alt="image" src="https://github.com/Unipisa/Simu5G/assets/129278941/13b1c7d7-b777-4430-9ca9-5f31cb34fc90">
<img width="488" alt="image" src="https://github.com/Unipisa/Simu5G/assets/129278941/4dabb9a7-55af-46b7-84c8-0fa4d83cb157">

**Comment by Y-AMDIAZ** [comment]
??

**Comment by Y-AMDIAZ** [comment]
?? help

**Comment by Y-AMDIAZ** [comment]
There are no one can help me ?

**Comment by giovanninardini** [comment]
I am not sure I can understand what you want to do. 
I believe that you want to use, e.g., 802.11 to send data from a car to the RSU (and this is not related to Simu5G). Then, maybe you want to let the RSU send data to the eNB using radio 4G/5G interface? To do this, you must add a `cellularNic` to the RSU module, as it is a UE of the 4G/5G network (see the Ue.ned module)

**Comment by Y-AMDIAZ** [comment]
![image](https://github.com/Unipisa/Simu5G/assets/129278941/37c27494-23ec-4eb4-8d7d-e7f5e0ac3e0d)
![image](https://github.com/Unipisa/Simu5G/assets/129278941/f5a56191-03f9-49cf-a90b-2b4d18c98704)
Ue.ned module you mean the eNodeB.ned

**Comment by Y-AMDIAZ** [comment]
Yes, I aim to establish communication from RSUs to eNodeBs. When integrating RSUs into Simu5G, I'm aware that communication setups are crucial, but uncertain about where to implement them. You mentioned adding a 'cellularNic' component in the RSU.ned file. Additionally, when you referred to 'Ue.ned,' you meant eNodes, correct? My goal is to enable cars to send messages to both RSUs and eNodeBs, with RSUs relaying the messages to eNodeBs. Thus, I'm looking to establish communication between cars, RSUs, and eNodeBs
( switch between 802.11p and LTE-V2X interfaces)
